src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -224,7 +228,21 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
             // '/out' parameter has to have an existing directory.
             Directory.CreateDirectory(Path.GetDirectoryName(_outputR2RImage));
 
+            WarningsDetected = false;
+
             return base.ExecuteTool(pathToTool, responseFileCommands, commandLineCommands);
         }
+
+        protected override void LogEventsFromTextOutput(string singleLine, MessageImportance messageImportance)
+        {
+            if (!ShowCompilerWarnings && singleLine.IndexOf("warning:", StringComparison.OrdinalIgnoreCase) != -1)
+            {
+                WarningsDetected = true;
```

Comment:
I think you should `Log.LogMessage` (at low importance?) even in warning-suppression mode. Dropping the warning lines entirely is surprising; demoting them in importance seems reasonable.

Reply:
Good call. Then we can always dig everything out of the binlog.

Reply:
That's a good idea. I'll make that change

Reply:
I kept the message with normal importance. It hides during the default verbosity, and for a higher verbosity level they don't show up in yellow (unless the property is set)